WordPress/Classes/Utility/ImmuTable.swift

```diff
@@ -0,0 +1,206 @@
+import Foundation
+import UIKit
+
+public typealias ImmuTableActionType = (ImmuTableRow) -> Void
+
+public protocol Reusable {
+    static var reusableIdentifier: String { get }
+}
+
+extension Reusable {
+    var reusableIdentifier: String {
+        get {
+            return self.dynamicType.reusableIdentifier
+        }
+    }
+}
+
+public protocol ImmuTableRow: Reusable {
+    var action: ImmuTableActionType? { get }
+    static var cellClass: AnyClass { get }
+    func configureCell(cell: UITableViewCell)
+    static var registrable: CellRegistrable { get }
+    static var customHeight: Float? { get }
+}
+
+public struct ImmuTableSection {
+    let headerText: String?
```

Comment:
I think for most cases I can think of, a plain String will be enough. If it becomes necessary we can add it later